internal/services/maintenance/public_maintenance_configurations_data_source.go

```diff
@@ -0,0 +1,191 @@
+					recurMondayToThursday,
+					recurFridayToSunday,
+				}, false),
+			},
+
+			"public_maintenance_configurations": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"location": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"duration": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"maintenance_scope": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"time_zone": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+						"recur_every": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourcePublicMaintenanceConfigurationsRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.PublicConfigurationsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resp, err := client.List(ctx)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("no Public Maintenance Configurations were found")
+		}
+		return fmt.Errorf("retrieving Public Maintenance Configurations: %+v", err)
+	}
+
+	filteredPublicConfigs := make([]interface{}, 0)
+
+	recurEveryFilter := d.Get("recur_every_filter").(string)
+	if recurEveryFilter == recurFridayToSunday {
+		recurEveryFilter = "week Friday, Saturday, Sunday"
+	} else if recurEveryFilter == recurMondayToThursday {
+		recurEveryFilter = "week Monday, Tuesday, Wednesday, Thursday"
+	}
+
+	locationFilter := d.Get("location_filter").(string)
+	scopeFilter := d.Get("scope_filter").(string)
+
+	for _, maintenanceConfig := range *resp.Value {
+
+		var configLocation, configRecurEvery, configScope string
+		if maintenanceConfig.Location != nil {
+			configLocation = *maintenanceConfig.Location
+		}
+		if props := maintenanceConfig.ConfigurationProperties; props != nil {
+			if props.Window != nil && props.Window.RecurEvery != nil {
+				configRecurEvery = *props.Window.RecurEvery
+			}
+			if string(props.MaintenanceScope) != "" {
+				configScope = string(props.MaintenanceScope)
+			}
+		}
+
+		if locationFilter == "" || locationFilter == configLocation {
+			if recurEveryFilter == "" || recurEveryFilter == configRecurEvery {
+				if scopeFilter == "" || scopeFilter == configScope {
+					filteredPublicConfigs = append(filteredPublicConfigs, flattenPublicMaintenanceConfiguration(maintenanceConfig))
+				}
+			}
+		}
+	}
+
+	d.Set("public_maintenance_configurations", filteredPublicConfigs)
```

Comment:
we'll also need to raise an error at this point if no maintainence configurations are returned:

```suggestion
	if len(filteredPublicConfigs) == 0 {
		return fmt.Errorf("no Public Maintenance Configurations were found")
	}
	if err := d.Set("public_maintenance_configurations", filteredPublicConfigs); err != nil {
		return fmt.Errorf("setting `public_maintenance_configurations`: %+v", err)
	}
```